nsot/models.py

```diff
@@ -1718,6 +1719,99 @@ def to_dict(self):
         }
 
 
+class Protocol(Resource):
+    """
+    Represetation of a routing protocol running over a circuit
+    """
+
+    device = models.ForeignKey(
+        Device, db_index=True, null=False, related_name='protocols',
+        verbose_name='Device that this protocol is running on'
+    )
+    circuit = models.ForeignKey(
+        Circuit, db_index=True, null=False, related_name='protocols',
+        verbose_name='Circuit that this protocol is running over',
+    )
+    site = models.ForeignKey(
+        Site, db_index=True, related_name='protocols',
+        on_delete=models.PROTECT,
+        help_text='Unique ID of the Site this Protocol is under.'
+    )
+
+    type = models.CharField(
+        max_length=8, choices=settings.PROTOCOL_TYPE_CHOICES, db_index=True
+    )
+    asn = models.PositiveIntegerField(db_index=True)
```

Comment:
Seems to me that `asn` here is specific only to BGP. I feel like if we're supporting multiple protocol types that this could be better named more abstractly? Curious what your thoughts on this are how it would related to IS-IS or OSPF, if at all?

Reply:
This was meant to also represent an area number is OSPF or a IS-IS level, but from our offline conversation there's not really a good field name to represent this generic level/area/ASN value, so this sort of thing is better suited for an attribute.